### ISSUE TITLE
Inconsistency in number for top performers products sold

### ISSUE THREAD

**Issue by rezzap** [woocommerce/woocommerce-ios#2386]
**Describe the bug**

A user has reported that they are seeing different numbers between the app and the browser version of their site under the number of products sold. The desktop accurately describes the number, but the app is inaccurate. 

Reported in 3026409-zen

**To Reproduce**
1. Go to My Store in the app
2. Scroll down to Top Performers
3. Compare the number listed here with the number seen under WC Admin dashboard in a browser for Top Products. 

The user believes this is related to a failed order which was then completed adding double the number, but we're not replicating this issue specifically and the inconsistency is present even before a failed order is added. A failed order is reflected on the dashboard in both the app and WC Admin right away when attempted on our end.

We tried a full sync on the user's site, but this did not change the numbers they are seeing in the app. 

**Screenshots**
@jaclync was able to reproduce this as discussed in p1591259522480600-slack-C6H8C3G23
<img width="1167" alt="Screen Shot 2020-06-04 at 4 47 03 PM" src="https://user-images.githubusercontent.com/10325627/83737049-66296100-a64a-11ea-90b6-0689bb7c78b2.png">

**Expected behavior**
I would expect the numbers in the app and the numbers in the browser to match. 

**Isolating the problem (mark completed items with an [x]):**
- [ ] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [ ] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [ ] I can reproduce this bug consistently using the steps above.

**Mobile Environment**
Please include:
- Device: iPhone X (my device where I replicated the user's issue)
- iOS version: 13.4.1
- WooCommerce iOS version: 4.3.0.2

**WordPress Environment**
<details>
### WordPress Environment ###

WordPress address (URL): https://shamory.com
Site address (URL): https://shamory.com
WC Version: 4.1.1
REST API Version: ✔ 1.0.7
WC Blocks Version: ✔ 2.5.16
Action Scheduler Version: ✔ 3.1.6
WC Admin Version: ✔ 1.1.3
Log Directory Writable: ✔
WP Version: 5.4.1
WP Multisite: –
WP Memory Limit: 768 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: Apache
PHP Version: 7.3.18
PHP Post Max Size: 128 MB
PHP Time Limit: 120
PHP Max Input Vars: 3000
cURL Version: 7.66.0
OpenSSL/1.1.1d-fips

SUHOSIN Installed: –
MySQL Version: 5.7.25-28-log
Max Upload Size: 128 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 4.1.1
WC Database Prefix: wp_
Total Database Size: 10.73MB
Database Data Size: 8.68MB
Database Index Size: 2.05MB
wp_woocommerce_sessions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_api_keys: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_attribute_taxonomies: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_downloadable_product_permissions: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_order_items: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_order_itemmeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_tax_rates: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_woocommerce_tax_rate_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zones: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_shipping_zone_locations: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_shipping_zone_methods: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_woocommerce_payment_tokens: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_woocommerce_payment_tokenmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_woocommerce_log: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_actions: Data: 0.36MB + Index: 0.17MB + Engine InnoDB
wp_actionscheduler_claims: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_groups: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_actionscheduler_logs: Data: 0.23MB + Index: 0.17MB + Engine InnoDB
wp_cmplz_cookiebanners: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_cmplz_cookies: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_cmplz_dnsmpd: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_cmplz_services: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_commentmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_comments: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mailchimp_carts: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mailchimp_jobs: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mc4wp_log: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo_campaign_log: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo_campaign_logmeta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo_conversions: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo_email_campaignmeta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo_email_campaigns: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_mo_optin_campaignmeta: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_mo_optin_campaigns: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_options: Data: 4.05MB + Index: 0.14MB + Engine InnoDB
wp_postmeta: Data: 0.30MB + Index: 0.17MB + Engine InnoDB
wp_posts: Data: 2.36MB + Index: 0.06MB + Engine InnoDB
wp_termmeta: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_terms: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_term_relationships: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_term_taxonomy: Data: 0.05MB + Index: 0.03MB + Engine InnoDB
wp_usermeta: Data: 0.06MB + Index: 0.03MB + Engine InnoDB
wp_users: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_admin_notes: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_admin_note_actions: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_category_lookup: Data: 0.02MB + Index: 0.00MB + Engine InnoDB
wp_wc_customer_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_download_log: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_coupon_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_order_product_lookup: Data: 0.02MB + Index: 0.06MB + Engine InnoDB
wp_wc_order_stats: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_wc_order_tax_lookup: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_wc_product_meta_lookup: Data: 0.02MB + Index: 0.09MB + Engine InnoDB
wp_wc_tax_rate_classes: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_wc_webhooks: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_indexable: Data: 0.09MB + Index: 0.05MB + Engine InnoDB
wp_yoast_indexable_hierarchy: Data: 0.02MB + Index: 0.05MB + Engine InnoDB
wp_yoast_migrations: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_primary_term: Data: 0.02MB + Index: 0.03MB + Engine InnoDB
wp_yoast_seo_links: Data: 0.02MB + Index: 0.02MB + Engine InnoDB
wp_yoast_seo_meta: Data: 0.05MB + Index: 0.00MB + Engine InnoDB

### Post Type Counts ###

attachment: 71
mc4wp-form: 1
nav_menu_item: 6
oembed_cache: 3
page: 11
post: 9
product: 1
revision: 254
shop_coupon: 22
shop_order: 27
shop_order_refund: 7
wpautoterms_page: 2
wpcf7_contact_form: 1

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Active Plugins (20) ###

WP AutoTerms: by WP AutoTerms – 2.4.0
BTCPay for WooCommerce: by BTCPay – 3.0.10 – Installed version not tested with active version of WooCommerce 4.1.1
Complianz | GDPR/CCPA Cookie Consent: by RogierLankhorst
Complianz – 4.6.2

Contact Form 7: by Takayuki Miyoshi – 5.1.9
Jetpack by WordPress.com: by Automattic – 8.5
MC4WP: Mailchimp for WordPress: by ibericode – 4.7.7
MC4WP: Mailchimp Top Bar: by ibericode – 1.5.2
MailOptin - Lite: by MailOptin Team – 1.2.28.1
MC4WP: Mailchimp for WordPress Premium: by ibericode – 4.8
SG Optimizer: by SiteGround – 5.5.5
Open Graph and Twitter Card Tags: by Webdados – 2.3.3
WooCommerce Admin: by WooCommerce – 1.1.3
WooCommerce PayPal Checkout Gateway: by WooCommerce – 2.0.2
WooCommerce Stripe Gateway: by WooCommerce – 4.4.0 – Installed version not tested with active version of WooCommerce 4.1.1
WooCommerce Services: by Automattic – 1.23.0 – Installed version not tested with active version of WooCommerce 4.1.1
WooCommerce USPS Shipping: by WooCommerce – 4.4.43
WooCommerce: by Automattic – 4.1.1
Yoast SEO Premium: by Team Yoast – 14.2
WordPress Starter: by SiteGround – 1.1.1
Yoast SEO: WooCommerce: by Team Yoast – 13.2

### Inactive Plugins (1) ###

Google Analytics for WordPress by MonsterInsights: by MonsterInsights – 7.10.4

### Settings ###

API Enabled: –
Force SSL: –
Currency: USD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: ✔

### WC Pages ###

Shop base: #43 - /shop/
Cart: #44 - /cart/
Checkout: #45 - /checkout/
My account: #46 - /my-account/
Terms and conditions: #445 - /wpautoterms/terms-and-conditions/

### Theme ###

Name: Flatsome
Version: 3.11.3
Author URL: http://www.uxthemes.com/
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
WooCommerce Support: ✔

### Templates ###

Overrides: flatsome/woocommerce/archive-product.php
flatsome/woocommerce/cart/cart-empty.php
flatsome/woocommerce/cart/cart-shipping.php
flatsome/woocommerce/cart/cart.php
flatsome/woocommerce/checkout/form-checkout.php
flatsome/woocommerce/checkout/form-coupon.php
flatsome/woocommerce/checkout/terms.php
flatsome/woocommerce/checkout/thankyou.php
flatsome/woocommerce/content-product.php
flatsome/woocommerce/content-product_cat.php
flatsome/woocommerce/content-single-product.php
flatsome/woocommerce/content-widget-product.php
flatsome/woocommerce/global/breadcrumb.php
flatsome/woocommerce/global/quantity-input.php
flatsome/woocommerce/global/wrapper-end.php
flatsome/woocommerce/global/wrapper-start.php
flatsome/woocommerce/loop/loop-end.php
flatsome/woocommerce/loop/loop-start.php
flatsome/woocommerce/loop/pagination.php
flatsome/woocommerce/loop/result-count.php
flatsome/woocommerce/loop/sale-flash.php
flatsome/woocommerce/myaccount/form-login.php
flatsome/woocommerce/myaccount/navigation.php
flatsome/woocommerce/notices/error.php
flatsome/woocommerce/notices/notice.php
flatsome/woocommerce/notices/success.php
flatsome/woocommerce/product-searchform.php
flatsome/woocommerce/single-product/photoswipe.php
flatsome/woocommerce/single-product/price.php
flatsome/woocommerce/single-product/product-image.php
flatsome/woocommerce/single-product/product-thumbnails.php
flatsome/woocommerce/single-product/rating.php
flatsome/woocommerce/single-product/related.php
flatsome/woocommerce/single-product/review.php
flatsome/woocommerce/single-product/sale-flash.php
flatsome/woocommerce/single-product/share.php
flatsome/woocommerce/single-product/short-description.php
flatsome/woocommerce/single-product/tabs/tabs.php
flatsome/woocommerce/single-product/title.php
flatsome/woocommerce/single-product/up-sells.php
flatsome/woocommerce/single-product-reviews.php
flatsome/woocommerce/single-product.php

### Action Scheduler ###

Complete: 1,062
Oldest: 2020-05-01 22:33:23 -0700
Newest: 2020-06-01 21:37:57 -0700

Pending: 2
Oldest: 2020-06-01 22:37:57 -0700
Newest: 2020-06-01 22:37:57 -0700
</details>

**Comment by rezzap** [comment]
The same issue seems to be happening in Android: https://github.com/woocommerce/woocommerce-android/issues/2451 

"Adopting the WC Admin API would be the best action when it's ready"

**Comment by jaclync** [comment]
I briefly checked if WC Admin is currently using an API to get the data. It looks like the leaderboards endpoint (e.g. `https://public-api.wordpress.com/rest/v1.1/jetpack-blogs/{{site_id}}/rest-api/?json=true&path=/wc-analytics/leaderboards%26interval%3Dhour%26before%3D2020-12-31T15%3A59%3A59Z%26per_page%3D3%26after%3D2020-01-01T16%3A00%3A00Z`) is returning data for visual display in WC Admin. Example response:

```
{
    "data": [
        ...
        {
            "id": "products",
            "label": "Top Products - Items Sold",
            "headers": [
                {
                    "label": "Product"
                },
                {
                    "label": "Items Sold"
                },
                {
                    "label": "Net Sales"
                }
            ],
            "rows": [
                [
                    {
                        "display": "<a href='https://funtestingusa.wpcomstaging.com/wp-admin/admin.php?page=wc-admin&path=/analytics/products&filter=single_product&products=2238'>Fun manga series</a>",
                        "value": "Fun manga series"
                    },
                    {
                        "display": "129",
                        "value": 129
                    },
                    {
                        "display": "<span class=\"woocommerce-Price-amount amount\"><span class=\"woocommerce-Price-currencySymbol\">&euro;</span>218!5000</span>",
                        "value": 218.5
                    }
                ],
                [
                    {
                        "display": "<a href='https://funtestingusa.wpcomstaging.com/wp-admin/admin.php?page=wc-admin&path=/analytics/products&filter=single_product&products=845'>Iced chocolate</a>",
                        "value": "Iced chocolate"
                    },
                    {
                        "display": "31",
                        "value": 31
                    },
                    {
                        "display": "<span class=\"woocommerce-Price-amount amount\"><span class=\"woocommerce-Price-currencySymbol\">&euro;</span>376!6000</span>",
                        "value": 376.5999999999999
                    }
                ],
                [
                    {
                        "display": "<a href='https://funtestingusa.wpcomstaging.com/wp-admin/admin.php?page=wc-admin&path=/analytics/products&filter=single_product&products=846'>Variable - chocolate bar</a>",
                        "value": "Variable - chocolate bar"
                    },
                    {
                        "display": "25",
                        "value": 25
                    },
                    {
                        "display": "<span class=\"woocommerce-Price-amount amount\"><span class=\"woocommerce-Price-currencySymbol\">&euro;</span>373!4000</span>",
                        "value": 373.40000000000003
                    }
                ]
            ]
        }
    ]
}
```

For us to be able to use this endpoint, we probably need:
- a dedicated endpoint for fetching top selling products. the current leaderboards endpoint returns more than just the products
- right now the order of the products in WC Admin is by item count, and we want to sort them by sale amount in the app (a new param like `orderby`)

**Comment by shiki** [comment]
Finishing #2422 should solve this issue.

**Comment by shiki** [comment]
@rezzap  This should now be fixed with #2550 as long as the user is using WC 4.0 or greater. The previous top performers stats data were taken from WPCOM + Jetpack sync. The new leaderboard stats, to be released on WCiOS 4.7, uses WCAdmin's API so the data should identically match now.
